server/src/resolvers/getVariantsResolver/adapters/g4rdAdapter.ts

```diff
@@ -86,23 +89,55 @@ const getG4rdNodeQuery = async ({
       },
       {
         headers: { Authorization, 'Content-Type': 'application/json', Accept: 'application/json' },
-        httpsAgent: new https.Agent({ rejectUnauthorized: false }),
       }
     );
 
     // Get patients info
     if (G4RDVariantQueryResponse) {
-      const individualIds = G4RDVariantQueryResponse.data.results.map(
-        v => v.individual.individualId
-      );
-      const patientUrl = `${process.env.G4RD_URL}/rest/patients/fetch?${individualIds
-        .map(id => `id=${id}`)
-        .join('&')}`;
-
-      G4RDPatientQueryResponse = await axios.get<G4RDPatientQueryResult>(patientUrl, {
-        headers: { Authorization, 'Content-Type': 'application/json', Accept: 'application/json' },
-        httpsAgent: new https.Agent({ rejectUnauthorized: false }),
-      });
+      let individualIds = G4RDVariantQueryResponse.data.results
+        .map(v => v.individual.individualId!)
+        .filter(Boolean); // Filter out undefined and null values.
+
+      // Get all unique individual Ids.
+      individualIds = [...new Set(individualIds)];
+
+      if (individualIds.length > 0) {
+        const patientUrl = `${process.env.G4RD_URL}/rest/patients/fetch?${individualIds
+          .map(id => `id=${id}`)
+          .join('&')}`;
+
+        G4RDPatientQueryResponse = await axios.get<G4RDPatientQueryResult>(patientUrl, {
+          headers: {
+            Authorization,
+            'Content-Type': 'application/json',
+            Accept: 'application/json',
+          },
+        });
+
+        // Get Family Id for each patient.
+
+        const patientFamily = axios.create({
+          headers: {
+            Authorization,
+            'Content-Type': 'application/json',
+            Accept: 'application/json',
+          },
+        });
+
+        const familyResponses = await Promise.allSettled(
+          individualIds.map(id =>
+            patientFamily.get<G4RDFamilyQueryResult>(
+              `${process.env.G4RD_URL}/rest/patients/${id}/family`
```

Comment:
ditto, watch out for ids that may need escaping, unless there's a guarantee from the PhenoTips API that there's no problem